openedx/core/djangoapps/user_api/views.py

```diff
@@ -284,9 +284,9 @@ def get(self, request):
 
         # Add a data sharing consent dialog if it's requested
         if active_provider_requests_data_sharing(request):
-            required = active_provider_requires_data_sharing(request)
+            enforced = active_provider_enforces_data_sharing(request)
             provider_name = Registry.get_from_pipeline(pipeline.get(request)).name
-            self._add_data_sharing_consent_field(form_desc, provider_name, required)
+            self._add_data_sharing_consent_field(form_desc, provider_name, required=enforced)
```

Comment:
@mattdrayer argument `required` here corresponds to html attribute, that is why I did not want to rename it here, I have also added a comment here to explain this.